### PR TITLE
Mongoose support 

### PR DESCRIPTION
**Pull request by pacostas**

### CODE REVIEW

File: clients/MONGODB/mongodb.js [hunk 1]
```diff
@@ -9,27 +8,21 @@ module.exports = {
...
+
+    const encodedPassword = binding.connectionOptions.auth.password
+      ? encodeURIComponent(binding.connectionOptions.auth.password)
+      : '';
+
+    const srv = binding.srv === 'true';
+
+    binding.url = [
+      'mongodb',
+      srv ? '+srv' : '',
```
Comment (mhdawson) [inline clients/MONGODB/mongodb.js:23]: I don't think this woud support not having a userid, which I think might be possible and the old code supported?

Comment (pacostas) [reply clients/MONGODB/mongodb.js:23]: the userid is the encodeduser variable, which is part of the mongo uri

Comment (mhdawson) [reply clients/MONGODB/mongodb.js:23]: My point is that from https://www.mongodb.com/docs/manual/reference/connection-string/

the username:password is optional, so it is possible to connect without it.  The old code handed that but I don't think the modified code does.

Comment (mhdawson) [reply clients/MONGODB/mongodb.js:23]: We can address this comment in a follow on PR if necessary.

**Review by mhdawson — APPROVED** [review]
LGTM

**Pull request merged**